File: IS_Lab2/IS_Lab2_C#/Kod źródłowy/Convert_json_to_yaml.cs
namespace Lab2
{
    internal class Convert_json_to_yaml
    {
        public static void run(Deserialize_json deserializeData, string destinationFileLocaiton)
        {
            Console.WriteLine("let's convert JSON to Yaml");

            var yamlSerializer = new YamlDotNet.Serialization.Serializer();
            string yaml = yamlSerializer.Serialize(deserializeData.data);
            File.WriteAllText(destinationFileLocaiton, yaml);

            Console.WriteLine("it's done");
        }

        public static void run(string filename, string destinationFileLocaiton)
        {
            Console.WriteLine("let's convert JSON to Yaml with filepath");

            Deserialize_json deserializator = new Deserialize_json(filename);
            Convert_json_to_yaml.run(deserializator, destinationFileLocaiton);

            Console.WriteLine("it's done");
        }
    }
}

File: IS_Lab2/IS_Lab2_JSON/Kod źródłowy/convert_json_to_yaml.py
# -*- coding: utf-8 -*-
"""
json to yaml converter
"""
import yaml
from functools import singledispatch


class ConvertJsonToYaml:

    @singledispatch
    @staticmethod
    def run(deserializeddata, destinationfilelocaiton):
        print("let's convert JSON to Yaml")
        with open(destinationfilelocaiton, 'w', encoding='utf8') as f:
            yaml.dump(deserializeddata, f, allow_unicode=True)
        print("it is done\n")

    @run.register
    @staticmethod
    def _(filename: str, destinationfilelocaiton):
        print("let's convert JSON to Yaml with filepath")
        from deserialize_json import DeserializeJson
        deserializeddata = DeserializeJson(filename)

        with open(destinationfilelocaiton, 'w', encoding='utf8') as f:
            yaml.dump(deserializeddata, f, allow_unicode=True)
        print("it is done\n")
